pkg/operator/options/options.go

```diff
@@ -66,26 +91,96 @@ func New() *Options {
 	f.StringVar(&opts.LogLevel, "log-level", env.WithDefaultString("LOG_LEVEL", ""), "Log verbosity level. Can be one of 'debug', 'info', or 'error'")
 
-	if opts.MemoryLimit > 0 {
-		newLimit := int64(float64(opts.MemoryLimit) * 0.9)
-		debug.SetMemoryLimit(newLimit)
-	}
-	if !lo.Contains(validLogLevels, opts.LogLevel) {
-		log.Fatalf("invalid log level %q passed through environment variables or cli arguments", opts.LogLevel)
-	}
+	// Vars that must be merged with settings
+	f.DurationVar(&opts.BatchMaxDuration, "batch-max-duration", env.WithDefaultDuration("BATCH_MAX_DURATION", 10*time.Second), "The maximum length of a batch window. The longer this is, the more pods we can consider for provisioning at one time which usually results in fewer but larger nodes.")
+	f.DurationVar(&opts.BatchIdleDuration, "batch-idle-duration", env.WithDefaultDuration("BATCH_IDLE_DURATION", time.Second), "The maximum amount of time with no new pending pods that if exceeded ends the current batching window. If pods arrive faster than this time, the batching window will be extended up to the maxDuration. If they arrive slower, the pods will be batched separately.")
+	f.StringVar(&opts.FeatureGates.inputStr, "feature-gates", env.WithDefaultString("FEATURE_GATES", "Drift=false"), "Optional features can be enabled / disabled using feature gates. Current options are: Drift")
+
 	return opts
 }
 
-// MustParse reads the user passed flags, environment variables, and default values.
-// Options are valided and panics if an error is returned
-func (o *Options) MustParse() *Options {
-	err := o.Parse(os.Args[1:])
+func (o *Options) MustParse(args ...string) *Options {
+	if err := o.Parse(args); err != nil {
+		if errors.Is(err, flag.ErrHelp) {
+			os.Exit(0)
+		}
+		panic(err)
+	}
 
-	if errors.Is(err, flag.ErrHelp) {
-		os.Exit(0)
+	if !lo.Contains(validLogLevels, o.LogLevel) {
+		log.Fatalf("invalid log level %q passed through environment variables or cli arguments", o.LogLevel)
 	}
-	if err != nil {
-		panic(err)
+
+	o.FeatureGates = MustParseFeatureGates(o.FeatureGates.inputStr)
+
+	// Check if shared fields have been set. If they haven't, they may be ovewritten by settings parsed from configmaps.
+	o.Visit(func(f *flag.Flag) {
+		switch f.Name {
+		case "batch-max-duration":
+			o.BatchMaxDurationSet = true
+		case "batch-idle-duration":
+			o.BatchIdleDurationSet = true
+		case "feature-gates":
+			o.FeatureGatesSet = true
+		}
+	})
+	if _, ok := os.LookupEnv("BATCH_MAX_DURATION"); ok {
+		o.BatchMaxDurationSet = true
+	}
+	if _, ok := os.LookupEnv("BATCH_IDLE_DURATION"); ok {
+		o.BatchIdleDurationSet = true
 	}
+	if _, ok := os.LookupEnv("FEATURE_GATES"); ok {
+		o.FeatureGatesSet = true
+	}
+
 	return o
 }
+
+// MergeSettings applies settings specified in the v1alpha5 configmap to options. If the value was already specified by
+// a CLI argument or environment variable, that value will be used.
+func (o *Options) MergeSettings(s *settings.Settings) {
```

Comment:
Thinking about this a little more: I'm a little concerned about how someone calling this function from the outside would reason about this function mutating the options in place. What are your thoughts about doing something like

```go
func MergeSettingsInto(s *settings.Settings, o *Options) {
  // same logic here
}
```

This makes it clear that you've made a change

Reply:
I agree that this could be made more clear. Thoughts on something like this:
```go
func (o *Options) MergeSettings(s *settings.Settings) *options.Options {
  // ...
}

options.ToContext(ctx, options.MergeSettings(settings.FromContext(ctx))
```
I think I like this a little more since it's a consistent signature style with `MustParse` but I'm not against your suggestion either.

Reply:
Looks perfect to me! 🎉